src/2018/transitioning/modules/macros.md

```diff
@@ -41,6 +41,34 @@ fn main() {
 
 This moves `macro_rules` macros to be a bit closer to other kinds of items.
 
+
+## Procedural macros
+
+When using procedural macros to derive traits, you could have written:
+
+```rust,ignore
+// Rust 2015
+extern crate serde;
+#[macro_use] extern crate serde_derive;
+
+#[derive(Serialize, Deserialize)]
+struct Bar;
+```
+
+Now, you write instead:
+
+```rust,ignore
+// Rust 2018
+#![feature(rust_2018_preview, use_extern_macros)]
+use serde_derive::{Serialize, Deserialize};
+
+#[derive(Serialize, Deserialize)]
+struct Bar;
+```
+
+Note that the macros you have to import have the same name as the traits you want to derive.
```

Comment:
I don't think this is quite true, but rather the macro's name is defined by `#[proc_macro_derive(Foo)]` and then it's simply named `Foo`. It's good practice to match a trait but it's not required!

Reply:
Maybe we could add some text like:

> When importing a macro to use with custom derive, you will have to name the macro that provides the custom derive.  This generally matches the name of the trait, but check with the documentation of the crate providing the derives to be sure.  
>
> For example, here is how you would do it with serde **_insert example here_**

This has a subtle benefit of encouraging library authors to now document this :)

Reply:
Sounds good!